src/tax-lot/tax-lot.repo.ts

```diff
@@ -0,0 +1,175 @@
+import { Inject } from "@nestjs/common";
+import { DB, DbType } from "src/global/providers/db.provider";
+import { eq, sql } from "drizzle-orm";
+import { DataRetrievalException, ResourceNotFoundException } from "src/error";
+import {
+  SelectTaxLot,
+  SelectTaxLotNested,
+  SelectTaxLotSpatial,
+} from "src/schema/tax-lot";
+import {
+  taxLot,
+  zoningDistrict,
+  zoningDistrictClass,
+  zoningDistrictZoningDistrictClass,
+} from "src/schema";
+import { MultiPolygon } from "geojson";
+
+export class TaxLotRepo {
+  constructor(
+    @Inject(DB)
+    private readonly db: DbType,
+  ) {}
+
+  #checkTaxLotByBbl = this.db.query.taxLot
+    .findFirst({
+      columns: {
+        bbl: true,
+      },
+      where: (taxLot, { eq, sql }) => eq(taxLot.bbl, sql.placeholder("bbl")),
+    })
+    .prepare("checkTaxLotByBbl");
+
+  async taxLotByBbl(bbl: string) {
+    let result: SelectTaxLotNested | undefined;
+    try {
+      result = await this.db.query.taxLot.findFirst({
+        columns: {
+          boroughId: false,
+          landUseId: false,
+          wgs84: false,
+          liFt: false,
+        },
+        where: (taxLot, { eq }) => eq(taxLot.bbl, bbl),
+        with: {
+          borough: true,
+          landUse: true,
+        },
+      });
+    } catch {
+      throw DataRetrievalException;
+    }
+
+    if (result === undefined) throw ResourceNotFoundException;
+    return result;
```

Comment:
We'll want to keep this section in the service. It is concerned with the business logic of inspecting the data quality and deciding how to handle it.